phpBB/phpbb/composer/io/translate_composer_trait.php

```diff
@@ -0,0 +1,237 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\composer\io;
+
+use phpbb\language\language;
+use Symfony\Component\Console\Output\OutputInterface;
+
+trait translate_composer_trait
+{
+	/**
+	 * @var language
+	 */
+	protected $language;
+
+	/**
+	 * @var array
+	 */
+	protected $composer_error;
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function write($messages, $newline = true)
+	{
+		$messages = (array) $messages;
+		$translated_messages = [];
```

Comment:
PHP 5.4.0 only. Is that officially the new min. req. yet? I still see 5.3.9 as our min. req.